pkg/xds/topology/dataplanes.go

```diff
@@ -3,23 +3,22 @@ package topology
 import (
 	"context"
 
-	"github.com/kumahq/kuma/pkg/core/dns"
+	"github.com/kumahq/kuma/pkg/core/runtime"
 
 	"github.com/pkg/errors"
 
 	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
-	"github.com/kumahq/kuma/pkg/core/resources/manager"
 )
 
 // GetDataplanes returns list of Dataplane in provided Mesh and Ingresses (which are cluster-scoped, not mesh-scoped)
-func GetDataplanes(ctx context.Context, rm manager.ReadOnlyResourceManager, mesh string) (*core_mesh.DataplaneResourceList, error) {
+func GetDataplanes(rt runtime.RuntimeContext, ctx context.Context, mesh string) (*core_mesh.DataplaneResourceList, error) {
 	dataplanes := &core_mesh.DataplaneResourceList{}
-	if err := rm.List(ctx, dataplanes); err != nil {
+	if err := rt.ReadOnlyResourceManager().List(ctx, dataplanes); err != nil {
 		return nil, err
 	}
 	rv := &core_mesh.DataplaneResourceList{}
 	for _, d := range dataplanes.Items {
-		if err := ResolveAddress(d); err != nil {
+		if err := ResolveAddress(rt, d); err != nil {
 			return nil, err
```

Comment:
we should log and ignore such dataplane. Otherwise one dataplane with address that cannot be resolved will break the entire mesh